### ISSUE TITLE
Wrong behaviour of the compiler in the case with strict null check and optional chaining

### ISSUE THREAD

**Issue by Relz** [microsoft/TypeScript#35842]
**TypeScript Version:**  3.7.4

**Search Terms:** optional chaining, strict, strict-type-predicates

**Expected behavior:** compiler says "Object is possibly 'undefined'".

**Actual behavior:** compiler thinks that object can't be undefined => possible runtime error "Cannot read property of undefined".

**Code**
```ts
interface SomeObject {
    someProperty: unknown;
}

let lastSomeProperty: unknown | undefined;

function someFunction(someOptionalObject: SomeObject | undefined): void {
    if (someOptionalObject?.someProperty !== lastSomeProperty) {
        console.log(someOptionalObject); // Why compiler thinks that someOptionalObject can't be undefined now?
        console.log(someOptionalObject.someProperty); // Runtime error possible here
        lastSomeProperty = someOptionalObject?.someProperty;
    }
}

const someObject: SomeObject = {
    someProperty: 42
};

someFunction(someObject);
someFunction(undefined);

```

<details><summary><b>Output</b></summary>

```ts
"use strict";
let lastSomeProperty;
function someFunction(someOptionalObject) {
    var _a, _b;
    if (((_a = someOptionalObject) === null || _a === void 0 ? void 0 : _a.someProperty) !== lastSomeProperty) {
        console.log(someOptionalObject); // Why compiler thinks that someOptionalObject can't be undefined now?
        console.log(someOptionalObject.someProperty); // Runtime error possible here
        lastSomeProperty = (_b = someOptionalObject) === null || _b === void 0 ? void 0 : _b.someProperty;
    }
}
const someObject = {
    someProperty: 42
};
someFunction(someObject);
someFunction(undefined);

```

</details>

<details><summary><b>Compiler Options</b></summary>

```json
{
  "compilerOptions": {
    "noImplicitAny": true,
    "strictNullChecks": true,
    "strictFunctionTypes": true,
    "strictPropertyInitialization": true,
    "strictBindCallApply": true,
    "noImplicitThis": true,
    "noImplicitReturns": true,
    "useDefineForClassFields": false,
    "alwaysStrict": true,
    "allowUnreachableCode": false,
    "allowUnusedLabels": false,
    "downlevelIteration": false,
    "noEmitHelpers": false,
    "noLib": false,
    "noStrictGenericChecks": false,
    "noUnusedLocals": false,
    "noUnusedParameters": false,
    "esModuleInterop": true,
    "preserveConstEnums": false,
    "removeComments": false,
    "skipLibCheck": false,
    "checkJs": false,
    "allowJs": false,
    "declaration": true,
    "experimentalDecorators": false,
    "emitDecoratorMetadata": false,
    "target": "ES2017",
    "module": "ESNext"
  }
}
```

</details>

**Playground Link:** [Provided](http://www.typescriptlang.org/play/#code/JYOwLgpgTgZghgYwgAgMoHsC2EDyAjAKwgTGQG8AoZa5AZywgAUp0AHaMATwC5kBXEAGsQ6AO4gA3BQC+FCgBsIpeXFpgM2Zmw49+QkeOQAfPQBMIMUBFNSKMASWDoQdBgDEHYJyAAU9bDisXs5w8vhEJLwauITEpCYC5pYg1gCUvABu6MCm5FQ0wDDIfgyBwSCh4XEA-AB0-kws7FBcyACEALwdyCpq0VrNXKl5NKPICM70irXy6ADmJQFB3pWxJKkSyAD0W8gA6gAWnONYrMCKUMhgB6CCtFcHcKQNZStha6QIcCAA5KR4KESFisuQM1XyYxOICmEBm80WuGWIXeETA9QYAx0G22uwASgIvNhkNAWJdWOhaLRgHhFMgDtAIBCxr11BimjpkN0XkiKiiaujNOyWpwpKNZLIKBNoc9Sh8orLUZyRjQGpjhbwACwAJhktgaHhAjmcCKq6yk+s83h8QOSaSkQA)

**Comment by Berkmann18** [comment]
The problem is indeed the `===` when the [specs](https://github.com/tc39/proposal-optional-chaining/blob/master/README.md#semantics) use `==` which makes statements like `opts && opts.thing` semantically different to `opts?.thing`.

It would have made more sense to have (in your case):
```
lastSomeProperty = (_b = someOptionalObject) === null || _b === undefined || _b === void 0 ? void 0 : _b.someProperty
```
Or simply:
```
lastSomeProperty = (_b = someOptionalObject) == null || _b === void 0 ? void 0 : _b.someProperty
```

**Comment by jack-williams** [comment]
Fix at #35895 

For inequality tests it was looking for _maybe_ undefined when it should be _definitely_ undefined. The `unknown` type is maybe undefined, but not definitely.
